pkg/runtime/plan/show_index.go

```
@@ -49,14 +54,37 @@ func (s *ShowIndexPlan) ExecIn(ctx context.Context, conn proto.VConn) (proto.Res
 		err     error
 	)
 
+	toTable := s.Stmt.TableName().Suffix()
+
+	db, table := s.Shards.Smallest()
+	s.Stmt.SetTableName(ast.TableName{table})
+
 	if err = s.Stmt.Restore(ast.RestoreDefault, &sb, &indexes); err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	var (
-		query = sb.String()
-		args  = s.toArgs(indexes)
+	query, err := conn.Query(ctx, db, sb.String(), s.toArgs(indexes)...)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	ds, _ := query.Dataset()
+	fields, _ := ds.Fields()
```

Do not ignore error, should return err directly.